### PR TITLE
ENG-808: Add custom handler functions support to OXD profile pic & link cell types

### PR DESCRIPTION
**Pull request by DanuAnjana**
## Checklist

- [x] Test Coverage is 100% for the newly added code
- [x] Storybook stories are added/updated for the changed areas
- [x] Components standards defined [in this document](https://docs.google.com/document/d/16_Nd3VxE_lTD9pVkONQ0egn7IiwyX1pVXZjzl-V4tU8/) are followed
- [x] Code is linted properly
- [x] Developer testing is done for the affected areas
- [ ] Package version updated (not applicable to ent branch)
- [x] Changelog.md updated on possible breaking (applicable to ent branch)

### CODE REVIEW

File: components/src/core/components/ProfilePic/ProfilePic.vue [hunk 1]
```diff
@@ -55,6 +60,23 @@ export default defineComponent({
         return TARGETS.indexOf(value) !== -1;
       },
     },
+    header: {
```
Comment (Super-Chama) [inline components/src/core/components/ProfilePic/ProfilePic.vue]: This is not correct. ProfilePic component is not a part of the cardtable. Cardtable is just using it as component for its cells. hence headers and rowitems should not be passed here

Comment (Super-Chama) [reply components/src/core/components/ProfilePic/ProfilePic.vue]: If you must use a generic event to communicate the click to cell

File: components/src/core/components/ProfilePic/__tests__/profile-pic.spec.ts [hunk 2]
```diff
@@ -23,4 +23,44 @@ describe('ProfilePic.vue', () => {
...
+        link: 'https://orangehrm.com',
+      },
+    });
+
+    const link = wrapper.find('a');
+    const event = {preventDefault: jest.fn()};
+
+    await link.trigger('click', event);
+
+    expect(event.preventDefault).not.toHaveBeenCalled();
```
Comment (Super-Chama) [inline components/src/core/components/ProfilePic/__tests__/profile-pic.spec.ts]: I'm not sure if this assertion is correct?

File: changelog.md [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 2024-10-18 - 59c22c85b1256cc167408e93402f6fb85a3bb775 -Input/PasswordInput.vue - Updating password Input field
 
+2024-10-11 - 6d028db15ec82666c290476cc656467c1e95b9aa - LinkWithPill.vue, link-with-pill.spec.ts, ProfilePicture.vue, ProfilePic.vue, profile-pic.spec.ts, ListSkeleton.story.vue - add onclick event to prevent default behaviour of link and execute a function
+
```
Comment (Super-Chama) [inline changelog.md]: Best to update this with recent date because it might be difficult to track change

**Review by Super-Chama — CHANGES_REQUESTED** [review]
There are unrelated old commits here. best to rebase from upstream/ent

File: components/src/core/components/CardTable/Cell/ProfilePicture.vue [hunk 4]
```diff
@@ -12,6 +12,8 @@
     :link="profilePicture.link"
     :imageSrc="profilePicture.src"
     :link-mode="profilePicture.target"
+    :isCustomFunctionExist="isCustomFunctionExist"
```
Comment (Super-Chama) [inline components/src/core/components/CardTable/Cell/ProfilePicture.vue]: This is not the best way to do this. consider this approach.

Inside `components/CardTable/Cell/ProfilePicture.vue`

```html
<template>
...
  <oxd-profile-pic
    v-else
    :size="profilePicture.size"
    :link="profilePicture.link"
    :image-src="profilePicture.src"
    :link-mode="profilePicture.target"
    :link-handler="handleLinkClick"
  />
...
</template>
```

```ts
<script lang="ts">
...
   setup() {
     ...
     const handleLinkClick = (event: MouseEvent) => {
        const cellConfig = props.header?.cellConfig;
        if (cellConfig && typeof cellConfig?.onClick === 'function') {
          event.preventDefault();
          props.header?.cellConfig.onClick(props.rowItem, event);
        }
    };
     ...
   }
....
</script>
```

Inside `components/ProfilePic/ProfilePic.vue`

```ts
<script lang="ts">
...
export default defineComponent({
  props: {
    linkHandler: {
      type: Function,
      required: false
    }
  }
})
...
methods: {
    handleLinkClick(event: MouseEvent) {
      if (this.linkHandler && typeof this.linkHandler === "function") {
        this.linkHandler(event);
      }
    },
  }
...
```

File: components/src/core/components/Icon/icons.ts [hunk 5]
```diff
@@ -2117,6 +2117,11 @@ export const oxdRequired: icon = {
 </svg>`,
 };
 
+export const oxdExternalLink: icon = {
+  name: 'oxd-external-link',
+  value: `<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><g id="SVGRepo_bgCarrier" stroke-width="0"/><g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round"/><g id="SVGRepo_iconCarrier"><g id="Interface / External_Link"><path id="Vector" style="fill:none" d="M10.0002 5H8.2002C7.08009 5 6.51962 5 6.0918 5.21799C5.71547 5.40973 5.40973 5.71547 5.21799 6.0918C5 6.51962 5 7.08009 5 8.2002V15.8002C5 16.9203 5 17.4801 5.21799 17.9079C5.40973 18.2842 5.71547 18.5905 6.0918 18.7822C6.5192 19 7.07899 19 8.19691 19H15.8031C16.921 19 17.48 19 17.9074 18.7822C18.2837 18.5905 18.5905 18.2839 18.7822 17.9076C19 17.4802 19 16.921 19 15.8031V14M20 9V4M20 4H15M20 4L13 11" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/></g></g></svg>`,
```
Comment (Super-Chama) [inline components/src/core/components/Icon/icons.ts:2122]: Make sure to add currentColor attribute and in the icon gallery story, the icon should display in red color

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: components/src/core/components/Icon/icons.ts (hunk 5 above)
Comment (DanuAnjana) [reply components/src/core/components/Icon/icons.ts:2122]: icon is currently displayed in red color

File: components/src/core/components/ProfilePic/ProfilePic.vue [hunk 6]
```diff
@@ -29,6 +34,9 @@ import {
 import {defaultUser} from './images';
 export default defineComponent({
   name: 'oxd-profile-pic',
+
+  emits: ['linkClick'],
```
Comment (Super-Chama) [inline components/src/core/components/ProfilePic/ProfilePic.vue]: seems not used

**Review by Super-Chama — APPROVED** [review]

**Pull request merged**